source/parsed_grid_generator.cc

```diff
@@ -290,8 +267,60 @@ void ParsedGridGenerator<dim, spacedim>::create(Triangulation<dim,spacedim> &tri
 
 }
 
+template <int dim>
+void ParsedGridGenerator<dim,dim>::create(Triangulation<dim,dim> &tria)
+{
+  Assert(this->grid_name != "", ExcNotInitialized());
+  if (this->grid_name == "rectangle")
+    {
+      GridGenerator::subdivided_hyper_rectangle ( tria,
+                                                  this->un_int_vec_option_one,
+                                                  this->point_option_two,
+                                                  this->point_option_one,
+                                                  this->bool_option_one);
+    }
+  else if (this->grid_name == "unit_hyperball")
+    {
+      Point<dim> center;
+      double radius=1.;
+      GridGenerator::hyper_ball (tria, center, radius);
```

Comment:
couldn't we maintain the old interface without the need of *Base and using here Asser(dim == spacedim, ExcNotImplemented);?